src/main/java/com/aws/greengrass/util/platforms/windows/WindowsRunasProcess.java

```diff
@@ -433,18 +467,40 @@ private static void writefd(final FileDescriptor fd, final WinNT.HANDLE pointer)
      * Convert environment Map to lpEnvironment block format.
      *
      * @return environment block for starting a process
+     * @see <a href="https://docs.microsoft.com/en-us/windows/win32/api/userenv/nf-userenv-createenvironmentblock">docs</a>
      */
-    private String computeEnvironmentBlock() {
-        // TODO get the existing env vars for the user
-        // Passing lpEnvironment to CreateProcessWithLogonW will overwrite the entire env
-        // This is ugly but need to call "cmd /C set" first to get all user's existing env vars
-
-        // Add SystemRoot env var if exists. See comment:
-        // https://github.com/openjdk/jdk/blob/b17b821/src/java.base/windows/classes/java/lang/ProcessEnvironment.java#L309-L311
-        String systemRootVal = System.getenv(SYSTEM_ROOT);
-        if (systemRootVal != null) {
-            additionalEnv.put(SYSTEM_ROOT, systemRootVal);
-        }
-        return Advapi32Util.getEnvironmentBlock(additionalEnv);
+    private String computeEnvironmentBlock(WinNT.HANDLE userTokenHandle) throws ProcessCreationException {
+        PointerByReference lpEnv = new PointerByReference();
+        // Get user's env vars, inheriting current process env
+        if (!UserEnv.INSTANCE.CreateEnvironmentBlock(lpEnv, userTokenHandle, true)) {
```

Comment:
If we're calling this, do we still need the hack calling `set` that I just wrote?